lib/queries/batchUpsert.js

```diff
@@ -1,32 +1,43 @@
 import valueSubQuery from './helpers/valueSubQuery';
 import batchParameter from './helpers/batchParameter';
+import sanitizeIdentifier from './helpers/sanitizeIdentifier';
+import whereQuery from './helpers/whereQuery';
 
 export default ({
     table,
     primaryKey,
     writableCols,
     returnCols = ['*'],
+    permanentFilters = {},
 }) => (rows) => {
-    const columns = primaryKey.concat(writableCols.filter((f) => primaryKey.indexOf(f) === -1));
+    const columns = primaryKey.concat(writableCols.filter(f => primaryKey.indexOf(f) === -1));
     const getValueSubQuery = valueSubQuery(columns);
     const getParameter = batchParameter(columns);
 
     const returning = returnCols.join(', ');
 
-    const setQuery = writableCols.map((col) => `${col} = excluded.${col}`);
+    const setQuery = writableCols.map(col => `${col} = excluded.${col}`);
 
     const values = rows
         .map((_, index) => getValueSubQuery(index + 1))
         .reduce((result, value) => result.concat(`(${value})`), []);
 
     const parameters = getParameter(rows);
 
+    const permanentFiltersKeys = Object.keys(permanentFilters);
+    let where = '';
+    let permanentParameters = {};
+    if (permanentFiltersKeys.length) {
+        permanentParameters = sanitizeIdentifier(permanentFiltersKeys, permanentFilters);
```

Comment:
You do not need sanitize parameter, here, since it come from declaration and won't change, we can assume that the filter is correct.
So you can use permanentFIlters directly.